lib/bibliothecary/parsers/go.rb

```diff
@@ -7,9 +7,19 @@ class Go
       include Bibliothecary::Analyser
 
       GPM_REGEXP = /^(.+)\s+(.+)$/
+      GOMOD_REGEX = /^(.+)\s+(.+)$/
+      GOSUM_REGEX = /^(.+)\s+(.+)\s+(.+)$/
 
       def self.mapping
         {
+          match_filename("go.mod") => {
+            kind: 'manifest',
+            parser: :parse_go_mod
+          },
+          match_filename("go.sum") => {
+            kind: 'checksum',
```

Comment:
A new file `kind` will have some implications for callers/users

Reply:
I think `lockfile` might make sense after all: it has packages that don't show up in `go.mod`, and technically a checksum is like an additional lock on the version 🤷‍♂